### PR TITLE
Coerce string integer destinations to file descriptors

### PR DESCRIPTION
**Pull request by jsumners**
It's easy to accidentally write code like the following and expect it "to work":

```js
'use strict'

const pino = require('pino')
const transport = pino.transport({
  targets: [
    { target: 'pino/file', level: 'debug', options: { destination: '2' } },
    { target: 'pino/file', level: 'info', options: { destination: '1' } }
  ]
})

const log = pino({ level: 'debug' }, transport)

log.error('something bad')
log.info('it works')
```

Notice that the `destination` in each target is a string integer. The clear intention is that they write to `stderr` and `stdout` respectively. This pull request accounts for this.

### CODE REVIEW

File: docs/transports.md [hunk 1]
```diff
@@ -262,7 +262,7 @@ const transport = pino.transport({
 pino(transport)
 ```
 
-The `options.destination` property may also be a number to represent a file descriptor. Typically this would be `1` to write to STDOUT or `2` to write to STDERR. If `options.destination` is not set, it defaults to `1` which means logs will be written to STDOUT.
+The `options.destination` property may also be a number to represent a filedescriptor. Typically this would be `1` to write to STDOUT or `2` to write to STDERR. If `options.destination` is not set, it defaults to `1` which means logs will be written to STDOUT. If `options.destination` is a string integer, e.g. `'1'`, it will be coerced to a number and used as a file descriptor. If this is not desired, provide a full path, e.g. `/tmp/1`.
```
Comment (mcollina) [inline docs/transports.md:265]: The docs for pino.destination should be updated too.

**Comment by mojavelinux** [comment]
I don't understand why we need to coerce a string to a number. Isn't the differentiation between a file path and file descriptor the type? What if you want to write to a file that is named "1". Now you lose that ability. For logs, that's not an unreasonable scenario (though probably rare) since logs are often sequenced in a directory. Can't we make the user get the type right?

File: docs/transports.md (hunk 1 above)
Comment (jsumners) [reply docs/transports.md:265]: 🤦‍♂️ will do. I literally went back to bed after this.

**Comment by jsumners** [comment]
The included docs show how to write to a file named `'1'`.

**Comment by mojavelinux** [comment]
That just seems really weird to me that we are not accepting what the user is providing. Sure, we can try to analyze the value and figure whether it is one or the other, but the type was already doing that.

**Comment by jsumners** [comment]
No, it wasn't. This is a reduction in surprise. Passing a single integer as a destination is a mistake and most likely an attempt to write to stdout or stderr. If the string is accepted as-is, it is not defined what will actually happen. If it _does_ write to a file, then that file can be in the current working directory or somewhere else entirely. If a file is the intended destination then an actual file path should be specified.

**Comment by mojavelinux** [comment]
We can agree to disagree, but I think the current behavior is the logical one. A integer is a file descriptor (which might be stdout, stderr, or one the application is managing). A string is a file path. As a user of pino, that's how I understood it and that contract makes complete sense to me.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mojavelinux** [comment]
After sitting on this for awhile, I now agree this is consistent with the *nix convention. I retract my earlier objection. Thanks for making this improvement.

**Comment by jsumners** [comment]
> After sitting on this for awhile, I now agree this is consistent with the *nix convention. I retract my earlier objection. Thanks for making this improvement.

Thank you.

**Pull request merged**